src/hermes/model/path.py

```diff
@@ -245,7 +245,14 @@ def _set_item(self, target: dict | list, path: 'ContextPath', value: t.Any, **kw
                 match target[key]:
                     case dict() as item: item.update(value)
                     case list() as item: item[:] = value
-                    case _: target[key] = value
+                    case _:
+                        if target[key] != value:
+                            tag = kwargs.pop('tag', {})
+                            alt = tag.pop('alternatives', [])
+                            alt.append((target[key], tag.copy()))
+                            tag.clear()
+                            tag['alternatives'] = alt
+                        target[key] = value
```

Comment:
These lines (244-255) are an exact duplicate of the code above. Call a dedicated extracted function instead?